### PR TITLE
Delete multiple resource

### PR DESCRIPTION
**Pull request by evidolob**
This PR provides new implementation of 'delete' command, now it supports deleting multiple selected items in tekton tree.

Fix: #213

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -755,14 +607,14 @@
           "group": "c1@3"
         },
         {
-          "command": "tekton.pipeline.delete",
-          "when": "view =~ /^tekton(CustomTree|PipelineExplorer)View/  && viewItem == pipeline",
+          "command": "tekton.explorerView.delete",
+          "when": "view =~ /^tekton(PipelineExplorer)View/ && viewItem =~ /^(pipeline|pipelinerun|clustertask|task|pipelineresource|triggertemplates|triggerbinding|clustertriggerbinding|eventlistener|conditions)/",
```
Comment (sudhirverma) [inline package.json:611]: you should skip delete command on root node like pipeline, pipelineRuns.

**Comment by sudhirverma** [comment]
as a user, I am getting confused if I select pipelineRun and go on  ClusterTriggerBinding node to delete ClusterTriggerBinding item in warning message it suggests me to delete pipelineRun.

**Comment by evidolob** [comment]
> If I am trying to delete a particular resource without clicking on it from explorer. It's not able to find the context value for it. I think we should provide a context if are trying to delete the resource without clicking on a particular item.

That behaviour it a bit strange, I haven't see such in any IDE, as IDE user I want to control what I'm deleting.
We can have such, but for me, this feature will be very rarely used, and it is requires to have own delete command for each resource.
@siamaksade WDYT?

**Comment by sudhirverma** [comment]
Screenshot 2020-07-28 at 1 20 48 AM

I can't see the delete command for `taskrun` for the single resource and for multiple resources.

**Comment by sudhirverma** [comment]
delete command is not working for `ClusterTriggerBinding`.

**Comment by evidolob** [comment]
> delete command is not working for `ClusterTriggerBinding`.

I implement it, previously we do not have it

File: package.json [hunk 2]
```diff
@@ -348,18 +316,6 @@
...
         "enablement": "tekton:tkn"
       },
-      {
-        "command": "tekton.pipelinerun.delete",
-        "title": "Delete",
-        "category": "Tekton",
-        "enablement": "tekton:tkn"
-      },
-      {
-        "command": "tekton.pipelinerun.delete.palette",
```
Comment (sudhirverma) [inline package.json:358]: this will remove the functionality for deleting pipelinerun from the command palette.

File: src/commands/delete.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import { TriggerTemplate } from '../tekton/triggertemplate';
+import { TriggerBinding } from '../tekton/triggerbinding';
+import { EventListener } from '../tekton/eventlistener';
+import { ClusterTask } from '../tekton/clustertask';
+import { TaskRun } from '../tekton/taskrun';
+import { window } from 'vscode';
+import { Progress } from '../util/progress';
+import { ClusterTriggerBinding } from '../tekton/clustertriggerbunding';
+
+type deleteFn = (node: TektonItem) => Promise<string | undefined>;
```
Comment (sudhirverma) [inline src/commands/delete.ts:25]: we are not using deletefn type anywhere can we remove this line?

File: package.json (hunk 2 above)
Comment (sudhirverma) [reply package.json:358]: and we have other commands also for delete operation from the command palette which is being removed.

Comment (evidolob) [reply package.json:358]: We already discuss that.

**Pull request merged**